File: sach/de3338.cpp
//DE3338.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
#define ll unsigned long long
int main() {
    ll n;
    cin>>n;
    ll gt = 1, tong = 0;
    for(ll i=1;i<=n;i++){
        gt = gt * i % 20240131;
        tong = (tong + gt) % 20240131;
        if (gt==0) break;
    }
    cout<<tong;
    return 0;
}

File: sach/de3338.py
//DE3338.PY - Tran Huu Nam - hsgtin.vn - 2024
n = int(input())
MOD = 20240131
tich = 1
tong = 1
for i in range(2, n+1):
    tich = (tich * i) % MOD
    tong = (tong + tich) % MOD
    if tich == 0: break
print(tong)

